connectd/peer_exchange_initmsg.c

```diff
@@ -51,13 +61,34 @@ static struct io_plan *peer_init_received(struct io_conn *conn,
 	if (unlikely(is_unknown_msg_discardable(msg)))
 		return read_init(conn, peer);
 
-	if (!fromwire_init(tmpctx, msg, &globalfeatures, &features)) {
+	if (!fromwire_init(tmpctx, msg, &globalfeatures, &features, tlvs)) {
 		status_peer_debug(&peer->id,
 				  "bad fromwire_init '%s', closing",
 				  tal_hex(tmpctx, msg));
 		return io_close(conn);
 	}
 
+	/* BOLT-ef7c97c02b6fa67a1df1af30b3843eb576100ebd #1:
+	 * The receiving node:
+	 * ...
+	 *  - upon receiving `networks` containing no common chains
+	 *    - MAY fail the connection.
+	 */
+	if (tlvs->networks) {
+		if (!tlvs->networks->chains) {
+			status_peer_debug(&peer->id,
+			                  "bad networks TLV in init '%s', closing",
+			                  tal_hex(tmpctx, msg));
+			return io_close(conn);
```

Comment:
Note this is technically redundant, since it's covered elegently by the next part anyway.